agent/engine/docker_client_test.go

```diff
@@ -1338,3 +1338,96 @@ func TestCreateVolume(t *testing.T) {
 	assert.Equal(t, volumeResponse.Volume.Mountpoint, mountPoint)
 	assert.Nil(t, volumeResponse.Volume.Labels)
 }
+
+func TestInspectVolumeTimeout(t *testing.T) {
+	timeout := 10*time.Millisecond
+	mockDocker, client, _, _, _, done := dockerClientSetup(t)
+	defer done()
+
+	warp := make(chan time.Time)
+	wait := &sync.WaitGroup{}
+	wait.Add(1)
+	mockDocker.EXPECT().InspectVolume(gomock.Any()).Do(func(x interface{}) {
+		warp <- time.Now()
+		wait.Wait()
+		// Don't return, verify timeout happens
+		// TODO remove the MaxTimes by cancel the context passed to CreateVolume
+		// when issue #1212 is resolved
+	}).MaxTimes(1)
+	volumeResponse := client.InspectVolume("name", timeout)
+	assert.Error(t, volumeResponse.Error, "expected error for timeout")
+	assert.Equal(t, "DockerTimeoutError", volumeResponse.Error.(api.NamedError).ErrorName())
+	wait.Done()
+}
+
+func TestInspectVolumeError(t *testing.T) {
+	mockDocker, client, _, _, _, done := dockerClientSetup(t)
+	defer done()
+
+	mockDocker.EXPECT().InspectVolume(gomock.Any()).Return(nil, errors.New("some docker error"))
+	volumeResponse := client.InspectVolume("name", 1*time.Second)
+	assert.Equal(t, "CannotInspectVolumeError", volumeResponse.Error.(api.NamedError).ErrorName())
+}
+
+func TestInspectVolume(t *testing.T) {
+	mockDocker, client, _, _, _, done := dockerClientSetup(t)
+	defer done()
+
+	volumeName := "volumeName"
+
+	VolumeOutput := docker.Volume{
+		Name: volumeName,
+		Driver: "driver",
+		Mountpoint: "local/mount/point",
+	}
+
+	mockDocker.EXPECT().InspectVolume(volumeName).Return(&VolumeOutput, nil)
+
+	volumeResponse := client.InspectVolume(volumeName, inspectContainerTimeout)
+	assert.Nil(t, volumeResponse.Error)
+	assert.Equal(t, VolumeOutput.Driver, volumeResponse.Volume.Driver)
+	assert.Equal(t, VolumeOutput.Mountpoint, volumeResponse.Volume.Mountpoint)
+	assert.Nil(t, volumeResponse.Volume.Labels)
```

Comment:
Let's test all the fields by adding a label in the `VolumeOutput` and check it here. Also `VolumeOutput` should be start with lowercase?